### PR TITLE
add docs

### PR DESCRIPTION
**Pull request by claws**
This pull request adds documentation to the uvloop project. A live demonstration can be found [here](http://uvloop-demo.readthedocs.io/en/add_docs/). 

The documentation contains a main page containing a brief overview. Then there is a user guide (install and using uvloop), a developers guide (getting the source, building, testing, docs) and the API page. The makefile has been updated to make it easy to build the docs.

The documentation uses Sphinx and can be linked to [readthedocs](https://readthedocs.org/) where many Python packages host their documentation.

I have also updated the loop.pyx file with comments similar to their `asyncio` counterparts.

There is also a suite of images/logos which are purely placeholders until you come up with a logo for the project. I think they represent a minimal set of images needed for good docs (e.g. favicon, logo in a few sizes, etc).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,11 +1,12 @@
-.PHONY: compile clean all distclean test debug sdist clean-libuv sdist-upload sdist-libuv
+.PHONY: compile clean all distclean test debug sdist clean-libuv \
+	sdist-upload sdist-libuv docs
```
Comment (1st1) [inline Makefile]: I'm not sure it's OK to have a line break in `.PHONY`. Just add a second section like this:

```
.PHONY: compile clean all distclean test debug sdist clean-libuv
.PHONY: sdist-upload sdist-libuv docs
```

File: README.rst [hunk 2]
```diff
@@ -1,9 +1,19 @@
...
 
+.. image:: https://travis-ci.org/MagicStack/uvloop.svg?branch=master
+    :target: https://travis-ci.org/MagicStack/uvloop
+
+.. image:: https://img.shields.io/pypi/dm/uvloop.svg
+   :target: https://pypi.python.org/pypi/uvloop
+
+.. image:: https://img.shields.io/badge/docs-latest-brightgreen.svg?style=flat
+   :target: http://uvloop.readthedocs.org/en/latest/
+
```
Comment (1st1) [inline README.rst]: I don't like when READMEs have a lot of shields -- it's hard to see the really important ones.  Let's keep the current shields (2) as is, and provide a link to the docs site.

File: uvloop/loop.pyx [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# cython: language_level=3
+# cython: language_level=3, embedsignature=True
```
Comment (1st1) [inline uvloop/loop.pyx]: Does `embedsignature`work for you? I couldn't make `inspect.signature` produce anything meaningful for `Loop` methods :(

File: docs/user/index.rst [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+==========
+
+This section of the documentation provides information about how to install
+uvloop as well as how to use uvloop.
+
+
+Installation
+------------
+
+`uvloop` is available from PyPI. It requires Python 3.5 and Cython.
```
Comment (1st1) [inline docs/user/index.rst]: uvloop only needs Cython for manual builds. If you install it from PyPI -- only Python 3.5 is needed.

File: docs/user/index.rst [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+Use pip to install it.
+
+.. code-block:: console
+
+    $ pip install uvloop
+
+.. note::
+
+    Use the latest version of Cython to minimise the potential for build
+    issues.
```
Comment (1st1) [inline docs/user/index.rst]: Again, this is only relevant for dev builds. And I think I'll add a `requirements.txt` file, and maybe add a `make` target to setup the dev venv.

Comment (1st1) [inline docs/user/index.rst]: Ah, I see you've added the `requirements.txt` file. I think we only need `requirements.dev.txt`. Please use `pip freeze` to list concrete deps versions.

File: docs/index.rst [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+`uvloop` is a fast, drop-in replacement of the built-in asyncio event loop.
+`uvloop` is released under the MIT license.
+
+`uvloop` and asyncio, combined with the power of async/await in Python 3.5,
+makes it easier than ever to write high-performance networking code in Python.
+
+`uvloop` makes asyncio fast. In fact, it is at least 2x faster than nodejs,
+gevent, as well as any other Python asynchronous framework. The performance of
+uvloop-based asyncio is close to that of Go programs.
```
Comment (1st1) [inline docs/index.rst]: Would be nice to have a link to the [blog post](http://magic.io/blog/uvloop-blazing-fast-python-networking/) here.

File: docs/dev/index.rst [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+specify the ``PYTHONPATH`` so that the `uvloop` package can be resolved from
+down in the ``tests`` directory.
+
+.. code-block:: console
+
+    $ cd uvloop/tests
+    $ PYTHONPATH=$PWD/.. python -m unittest
+
+
+Individual Tests
```
Comment (1st1) [inline docs/dev/index.rst]: I'd recommend to use `py.test` for individual tests -- that's what I'm myself using.  For example:

`$ PYTHONPATH=. py.test -k test_signals_sigint_uvcode`

File: docs/dev/index.rst [hunk 8]
```diff
@@ -0,0 +1,139 @@
+Developers Guide
+================
+
+The project is hosted on `GitHub <https://github.com/MagicStack/uvloop>`_.
+and uses `Travis <https://travis-ci.org/MagicStack/uvloop>`_ for
+Continuous Integration.
+
+If you have found a bug or have an idea for an enhancement that would
+improve the library, use the `bug tracker <https://github.com/MagicStack/uvloop/issues>`_.
+
```
Comment (1st1) [inline docs/dev/index.rst]: I'd also add that we consider any deviation from asyncio behaviour as a bug.

File: requirements.txt [hunk 9]
```diff
@@ -0,0 +1 @@
+Cython
```
Comment (1st1) [inline requirements.txt]: We don't need this file

**Comment by 1st1** [comment]
@claws This is one impressive PR. Thank you very much for working on this. I've left a few review comments and recommendations.

One more thing: I don't think that uvloop needs a logo/favicon.  Text is fine and less distracting.  If it's absolutely required by Sphinx or RTD, I can make a nice text-only logo.

**Comment by claws** [comment]
Hmmm, I got into a bit of a state with the branch trying to rebase recent updates. I'm not exactly sure what happened but it just looks a mess now. I think I might create a new branch, copy in the appropriate changes and re-submit the pull request under a different PR.

**Pull request closed without merging**